src/resources/views/fields/date_picker.blade.php

```diff
@@ -70,7 +70,12 @@
                 var $existingVal = $field.val();
 
                 if( $existingVal.length ){
-                    preparedDate = new Date($existingVal).format($customConfig.format);
+                    // Passing an ISO-8601 date string to the Date constructor results in varying
+                    // behavior across browsers. Splitting and passing in parts of the date manually
+                    // gives us more defined behavior.
+                    // See https://stackoverflow.com/questions/2587345/why-does-date-parse-give-incorrect-results
+                    var parts = $existingVal.split('-')
+                    preparedDate = new Date(parts[0], parts[1]-1, parts[2]).format($customConfig.format);
```

Comment:
Although I'd love to just make IE disappear in its entirety, unfortunately people still seem to use IE11 (so let's not drop that support yet).